src/LuzFaltex.Extensions.DependencyInjection/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -0,0 +1,205 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+
+namespace LuzFaltex.Extensions.DependencyInjection.ServiceLookup;
+
+internal sealed class ServiceProviderEngineScope : IServiceScope, IServiceProvider, IAsyncDisposable, IServiceScopeFactory
+{
+    // For testing only
+    internal IList<object> Disposables => _disposables ?? (IList<object>)Array.Empty<object>();
+
+    private bool _disposed;
+    private List<object>? _disposables;
+
+    public ServiceProviderEngineScope(MutableServiceProvider provider, bool isRootScope)
```

Comment:
All instances of ``ServiceProvider`` was changed to ``MutableServiceProvider`` on this file.

Reply:
I forgot to mark a few other files because of a single line change to those as well.